File: 240229/줄 세우기/line-up-students.py
class Student:
    def __init__(self, h, w, n):
        self.h, self.w, self.n = h ,w, n

n = int(input())
students = []

for i in range(1, n + 1):
    h, w = tuple(map(int, input().split()))
    students.append(Student(h, w, i))

students.sort(key = lambda x: (-x.h, -x.w, x.n))

for s in students:
    print(s.h, s.w, s.n)

File: 240229/줄 세우기/line-up-students.js
const fs = require("fs");
let [n, ...data] = fs.readFileSync(0).toString().trim().split("\n");

const students = []
for(let i = 0; i < n; i++) {
    const [height, weight] = data[i].trim().split(" ")
    students.push({height: Number(height), weight: Number(weight), num: i + 1})
}

function comp(a,b) {
    if((a.height === b.height) && (a.weight === b.weight)) {
        return a.num - b.num;
    }
    if((a.height === b.height)) {
        return b.weight - a.weight;
    }

    return b.height - a.height;
}

students.sort(comp)

for(let s of students){
    console.log(`${s.height} ${s.weight} ${s.num}`)
}